components/Chips/src/MDCChipField.m

```diff
@@ -615,6 +615,25 @@ - (BOOL)textFieldShouldReturn:(UITextField *)textField {
 - (void)textFieldDidChange {
   [self deselectAllChips];
   [self createNewChipWithTextField:self.textField delimiter:MDCChipFieldDelimiterSpace];
+  CGRect lastChipFrame = self.chips.lastObject.frame;
+  if (self.mdf_effectiveUserInterfaceLayoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
+    lastChipFrame = MDFRectFlippedHorizontally(lastChipFrame, CGRectGetWidth(self.bounds));
+  }
+
+  if (!CGRectIsEmpty(lastChipFrame)) {
+    BOOL isTextTooWide = [self textInputDesiredWidth] >= [self availableWidthForTextInput];
+    BOOL isTextFieldOnSameLineAsChips =
+        CGRectGetMidY(self.textField.frame) == CGRectGetMidY(lastChipFrame);
+    if (isTextTooWide && isTextFieldOnSameLineAsChips) {
+      // The text is on the same line as the chips and doesn't fit
+      // Trigger layout to move the text field down to the next line
+      [self setNeedsLayout];
+    } else if (!isTextTooWide && !isTextFieldOnSameLineAsChips) {
+      // The text is on the line below the chips but can fit on the same line
+      // Trigger layout to move the text field up to the previous line
+      [self setNeedsLayout];
+    }
+  }
```

Comment:
I'm finding it hard to identify what's net-new in this PR vs what's a refactoring; is it possible to split this into two smaller PRs?

Reply:
Pulled the refactoring into https://github.com/material-components/material-components-ios/pull/9827